ale/drivers/selene_drivers.py

```diff
@@ -435,3 +414,318 @@ def sensor_model_version(self):
           ISIS sensor model version
         """
         return 1
+
+
+class KaguyaMiPds3NaifSpiceDriver(LineScanner, Pds3Label, NaifSpice, Driver):
+    """
+    Driver for a PDS3 Kaguya Multiband Imager (Mi) images. Specifically level2b2 Vis and Nir images.
+
+    NOTES
+    -----
+
+    * Kaguaya has adjusted values for some of its keys, usually suffixed with `CORRECTED_`.
+      These corrected values should always be preffered over the original values.
+    """
+
+
+    @property
+    def utc_start_time(self):
+        """
+        Returns corrected utc start time.
+
+        If no corrected form is found, defaults to the form specified in parent class.
+
+        Returns
+        -------
+        : str
+          Start time of the image in UTC YYYY-MM-DDThh:mm:ss[.fff]
+        """
+        return self.label.get('CORRECTED_START_TIME', super().utc_start_time)
+
+    @property
+    def utc_stop_time(self):
+        """
+        Returns corrected utc start time.
+
+        If no corrected form is found, defaults to the form specified in parent class.
+
+        Returns
+        -------
+        : str
+          Stop time of the image in UTC YYYY-MM-DDThh:mm:ss[.fff]
+
+        """
+
+        return self.label.get('CORRECTED_STOP_TIME', super().utc_stop_time)
+
+    @property
+    def base_band(self):
+        """
+        Which band the bands are registered to.
+        """
+        band_map = {
+            "MV1" : "MI-VIS1",
+            "MV2" : "MI-VIS2",
+            "MV3" : "MI-VIS3",
+            "MV4" : "MI-VIS4",
+            "MV5" : "MI-VIS5",
+            "MN1" : "MI-NIR1",
+            "MN2" : "MI-NIR2",
+            "MN3" : "MI-NIR3",
+            "MN4" : "MI-NIR4"
+        }
+        base_band = band_map[self.label.get("BASE_BAND")]
+        return base_band
+
+    @property
+    def instrument_id(self):
+        """
+        Id takes the form of LISM_<BASE_BAND> where <BASE_BAND> is which band
+        the bands were registered to.
+
+        Returns
+        -------
+        : str
+          instrument id
+        """
+
+        id = f"LISM_{self.base_band}"
+        return id
+
+    @property
+    def sensor_frame_id(self):
+        """
+        Returns the sensor frame id.  Depends on the instrument that was used to
+        capture the image.
+
+        Returns
+        -------
+        : int
+          Sensor frame id
+        """
+        spectra = self.base_band[3]
+        return spice.namfrm(f"LISM_MI_{spectra}_HEAD")
+
+    @property
+    def spacecraft_name(self):
+        """
+        Returns "MISSION_NAME" as a proxy for spacecraft_name.
+
+        No NAIF code exists for the spacecraft name 'SELENE-M.'  The NAIF code
+        exists only for 'SELENE' or 'KAGUYA' -- 'SELENE' is captured as
+        'MISSION_NAME'
+
+        Returns
+        -------
+        : str
+          mission name
+        """
+        return self.label.get('MISSION_NAME')
```

Comment:
This is a little different from Kaguya TC. MI labels do not have the keyword that TC does, or a keyword with the same value. This returns the full name of the spacecraft including spaces.